src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -153,7 +153,7 @@ private static bool GetLinqExpressionsBuiltWithIsInterpretingOnly()
         public static bool IsNotIntMaxValueArrayIndexSupported => s_largeArrayIsNotSupported.Value;
 
         public static bool IsAssemblyLoadingSupported => !IsNativeAot;
-        public static bool IsNonBundledAssemblyLoadingSupported => !IsAssemblyLoadingSupported && !IsMonoAOT;
+        public static bool IsAssemblyLoadingFromFileSupported => IsAssemblyLoadingSupported && !IsMonoAOT;
```

Comment:
I'm curious what's different between Native AOT and Mono's AOT that the former is able to support loading assemblies from files and the latter isn't?

Reply:
In full AOT for mono, assembly loading is allowed, but in the case I'm trying to skip, the exe assembly wasn't AOT'd and was not registered as a module on startup.  For NativeAOT, my understanding is everything compiles into a single binary. That means you can't load anything external.

I kept going back and forth on a proper name for the property.  I reverted back to what I originally had as I now think it's a little more accurate.

Reply:
In NativeAOT, Assembly.Load works too because that one is not actually loading anything new. The assembly was part of the app.

Assembly.LoadFrom or loading from byte array don't work and are blocked on the existing condition. Does it match your semantic?

Reply:
Close. I say that because tests like this pass:

https://github.com/dotnet/runtime/blob/46db4aa3c505ac36cf8d0d612d4161bb7239ed87/src/libraries/System.Runtime/tests/System/ActivatorTests.cs#L626-L648

Reply:
I'll leave it up to you, but I would expect that API to be unsupported in general. It might work in corner cases like in the test but it's a much easier story to say to customers that apis that load assemblies from files are not supported than to have an asterisk with when it is supported. Especially if there's perfectly good replacement apis like CreateInstance(string,string).

Apis that are unsupported don't need test coverage.

Reply:
Tagging @lambdageek in the event I didn't explain it quite right.